include/zephyr/drivers/can.h

```diff
@@ -212,7 +215,7 @@ struct can_filter {
 	 */
 	uint32_t mask         : 29;
 	/** Flags. @see @ref CAN_FILTER_FLAGS. */
-	uint8_t flags         : 3;
+	uint8_t flags;
```

Comment:
It's a bit unfortunate that we will increase the struct size by another word size for just this one bit. But on the other hand the switch to flags was exactly done for the purpose of being more extensible in the future, as you mentioned in #51361.

I was wondering if the flag should be enabled only for `CONFIG_CAN_FD_MODE=y` to lower the footprint for other use cases? Or would that introduce too much noise for little benefit?

Reply:
No, the flags should always be defined regardless of Kconfig settings. We were bound to hit a point where more than three bits were needed for the filter flags at some point, so I don't see any issue with this minor increase in RAM footprint.